exasol/nb_connector/slc/script_language_container.py

```diff
@@ -0,0 +1,212 @@
+            )
+
+    def deploy(self):
+        """
+        Deploys the current script-languages-container to the database and
+        stores the activation string in the Secure Configuration Storage.
+        """
+        bfs_params = {
+            k: self.secrets.get(v)
+            for k, v in [
+                ("bucketfs_host", CKey.bfs_host_name),
+                ("bucketfs_port", CKey.bfs_port),
+                ("bucketfs_user", CKey.bfs_user),
+                ("bucketfs_password", CKey.bfs_password),
+                ("bucketfs_name", CKey.bfs_service),
+                ("bucket", CKey.bfs_bucket),
+            ]
+        }
+
+        with current_directory(self.checkout_dir):
+            exaslct_api.deploy(
+                flavor_path=(str(self._flavor_path_rel),),
+                **bfs_params,
+                path_in_bucket=constants.PATH_IN_BUCKET,
+                release_name=self.language_alias,
+                output_directory=str(self.workspace.output_path),
+            )
+            container_name = f"{self.flavor}-release-{self.language_alias}"
+            result = exaslct_api.generate_language_activation(
+                flavor_path=str(self._flavor_path_rel),
+                bucketfs_name=bfs_params["bucketfs_name"],
+                bucket_name=bfs_params["bucket"],
+                container_name=container_name,
+                path_in_bucket=constants.PATH_IN_BUCKET,
+            )
+            alter_session_cmd = result[0]
+            re_res = re.search(
+                r"ALTER SESSION SET SCRIPT_LANGUAGES='(.*)'", alter_session_cmd
+            )
+            activation_key = re_res.groups()[0]
+            _, url = activation_key.split("=", maxsplit=1)
+            self.secrets.save(self._alias_key, f"{self.language_alias}={url}")
+
+    @property
+    def _alias_key(self):
+        return constants.SLC_ACTIVATION_KEY_PREFIX + self.language_alias
+
+    @property
+    def activation_key(self) -> str:
+        """
+        Returns the language activation string for the uploaded script-language-container.
+        Can be used in `ALTER SESSION` or `ALTER_SYSTEM` SQL commands to activate
+        the language of the uploaded script-language-container.
+        """
+        try:
+            return self.secrets[self._alias_key]
+        except AttributeError as ex:
+            raise SlcError(
+                "Secure Configuration Storage does not contains an activation key."
+            ) from ex
+
+    def append_custom_packages(self, pip_packages: list[PipPackageDefinition]):
+        """
+        Appends packages to the custom pip file.
+
+        Note: This method is not idempotent: Multiple calls with the same
+        package definitions will result in duplicate entries.
+        """
+        with open(self.custom_pip_file, "a") as f:
+            for p in pip_packages:
+                print(f"{p.pkg}|{p.version}", file=f)
+
+    @property
+    def docker_images(self) -> list[str]:
```

Comment:
If the function returns the docker image tags, maybe we should rename to
```suggestion
    def docker_image_tags(self) -> list[str]:
```